test/e2e/validators.go

```diff
@@ -18,253 +20,256 @@ import (
 
-func newOperatorEmulation(n string, id string, bSPath string, bPath string, bRPath string) *OperatorEmulation {
-	return &OperatorEmulation{
-		buildRun:                &operator.BuildRun{},
-		clusterBuildStrategy:    &operator.ClusterBuildStrategy{},
-		buildStrategy:           &operator.BuildStrategy{},
-		build:                   &operator.Build{},
-		namespace:               n,
-		identifier:              id,
-		buildStrategySamplePath: bSPath,
-		buildSamplePath:         bPath,
-		buildRunSamplePath:      bRPath,
+// cleanupOptions return a CleanupOptions instance.
+func cleanupOptions(ctx *framework.TestCtx) *framework.CleanupOptions {
+	return &framework.CleanupOptions{
+		TestContext:   ctx,
+		Timeout:       cleanupTimeout,
+		RetryInterval: cleanupRetryInterval,
 	}
-
 }
 
-func createNamespacedBuildStrategy(
-	t *testing.T,
-	ctx *framework.TestCtx,
-	f *framework.Framework,
-	testBuildStrategy *operator.BuildStrategy) {
-	err := f.Client.Create(goctx.TODO(), testBuildStrategy, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
-	if err != nil {
+// createPipelineServiceAccount make sure the "pipeline" SA is created, or already exists.
+func createPipelineServiceAccount(t *testing.T, ctx *framework.TestCtx, f *framework.Framework) {
+	ns, err := ctx.GetNamespace()
+	require.NoError(t, err, "unable to obtain test namespace")
+	serviceAccount := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: ns,
+			Name:      TestServiceAccountName,
+		},
+	}
+	t.Logf("Creating '%s' service-account", TestServiceAccountName)
+	err = f.Client.Create(goctx.TODO(), serviceAccount, cleanupOptions(ctx))
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
 		t.Fatal(err)
 	}
 }
 
-func createClusterBuildStrategy(
-	t *testing.T,
-	ctx *framework.TestCtx,
-	f *framework.Framework,
-	testBuildStrategy *operator.ClusterBuildStrategy) {
-	err := f.Client.Create(goctx.TODO(), testBuildStrategy, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
-	if err != nil {
-		t.Fatal(err)
+// createContainerRegistrySecret use environment variables to check for container registry
+// credentials secret, when not found a new secret is created.
+func createContainerRegistrySecret(t *testing.T, ctx *framework.TestCtx, f *framework.Framework) {
+	secretName := os.Getenv(EnvVarImageRepoSecret)
+	secretPayload := os.Getenv(EnvVarSourceRepoSecretJSON)
+	if secretName == "" && secretPayload == "" {
+		t.Logf("Environment variable with container registry secret is not present.")
+		return
 	}
+
+	ns, err := ctx.GetNamespace()
+	require.NoError(t, err, "unable to obtain test namespace")
+
+	secretNsName := types.NamespacedName{Namespace: ns, Name: secretName}
+	secret := &corev1.Secret{}
+	if err = f.Client.Get(goctx.TODO(), secretNsName, secret); err == nil {
+		t.Logf("Container registry secret is found at '%s/%s'", ns, secretName)
```

Comment:
If the `EnvVarImageRepoSecret` is set, but `EnvVarSourceRepoSecretJSON` isn´t and the `secret` does not exists in the namespace, the logic will try to create a secret from a payload that was not provided. I just wanted to highlight this, because on most of the test code we are not catching errors. We can improve on this once we try the `parallel` implementation.

Reply:
Good catch! Let me improve this, thanks.

Reply:
Changing the logic to require both secret-name and payload (JSON), in order to create the secret. So, when either `TEST_IMAGE_REPO_SECRET` or `TEST_IMAGE_REPO_DOCKERCONFIGJSON` is empty, the secret is not created.

Reply:
> We can improve on this once we try the `parallel` implementation.

Indeed, in the parallel runs we might want to transform each test-case in a `t.Run()` call.

Reply:
nice!